File: CS/MyBrowser.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ChildStandaloneForm
{
    public partial class MyBrowser : UserControl
    {
        const int topIndent = 30;
        const string startPage = "http://www.google.com";
        public MyBrowser()
        {
            InitializeComponent();
            SubscribeWebBrowserEvents();
            LoadNewDocument(startPage);
            buttonEdit1.KeyDown += new KeyEventHandler(buttonEdit1_KeyDown);
            this.Resize += new EventHandler(MyBrowser_Resize);
            this.Disposed += new EventHandler(MyBrowser_Disposed);
        }

        void MyBrowser_Disposed(object sender, EventArgs e)
        {
            webBrowser1.Dispose();
        }

        void MyBrowser_Resize(object sender, EventArgs e)
        {
            webBrowser1.Width = this.Width - 6;
            webBrowser1.Height = this.Height - topIndent;
        }

        void buttonEdit1_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
                LoadNewDocument(buttonEdit1.Text);
        }

        private void SubscribeWebBrowserEvents()
        {
            webBrowser1.DocumentTitleChanged += new EventHandler(webBrowser1_DocumentTitleChanged);
            webBrowser1.CanGoBackChanged += new EventHandler(webBrowser1_CanGoBackChanged);
            webBrowser1.CanGoForwardChanged += new EventHandler(webBrowser1_CanGoForwardChanged);
        }

        void webBrowser1_CanGoForwardChanged(object sender, EventArgs e)
        {
            btnForward.Enabled = webBrowser1.CanGoForward;
        }

        void webBrowser1_CanGoBackChanged(object sender, EventArgs e)
        {
            btnBack.Enabled = webBrowser1.CanGoBack;
        }
        void webBrowser1_DocumentTitleChanged(object sender, EventArgs e)
        {
            SetTitle(webBrowser1.DocumentTitle);
            buttonEdit1.Text = Convert.ToString(webBrowser1.Url);
        }

        // Fields...
        private string _Title;

        public string Title { get { return _Title; } }

        public event EventHandler TitleChanged;

        protected virtual void OnTitleChangedEvent()
        {
            if (TitleChanged != null)
                TitleChanged(this, EventArgs.Empty);
        }

        private void SetTitle(string s)
        {
            _Title = s;
            OnTitleChangedEvent();
        }

        private void LoadNewDocument(string address)
        {
            if (String.IsNullOrEmpty(address)) return;
            if (address.Equals("about:blank")) return;
            if (!address.StartsWith("http://") &&
                !address.StartsWith("https://"))
            {
                address = "http://" + address;
            }
            try
            {
                webBrowser1.Navigate(new Uri(address));
                buttonEdit1.Text = address;
            }
            catch (System.UriFormatException)
            {
                return;
            }
        }

        private void buttonEdit1_ButtonClick(object sender, DevExpress.XtraEditors.Controls.ButtonPressedEventArgs e)
        {
            LoadNewDocument(buttonEdit1.Text);
        }

        private void simpleButton1_Click(object sender, EventArgs e)
        {
            webBrowser1.GoBack();
        }

        private void simpleButton2_Click(object sender, EventArgs e)
        {
            webBrowser1.GoForward();
        }

        private void simpleButton1_Click_1(object sender, EventArgs e)
        {
            LoadNewDocument(buttonEdit1.Text);
        }
    }
}

File: VB/MyBrowser.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace ChildStandaloneForm

    Public Partial Class MyBrowser
        Inherits UserControl

        Const topIndent As Integer = 30

        Const startPage As String = "http://www.google.com"

        Public Sub New()
            InitializeComponent()
            SubscribeWebBrowserEvents()
            LoadNewDocument(startPage)
            AddHandler buttonEdit1.KeyDown, New KeyEventHandler(AddressOf buttonEdit1_KeyDown)
            AddHandler Resize, New EventHandler(AddressOf MyBrowser_Resize)
            AddHandler Disposed, New EventHandler(AddressOf MyBrowser_Disposed)
        End Sub

        Private Sub MyBrowser_Disposed(ByVal sender As Object, ByVal e As EventArgs)
            webBrowser1.Dispose()
        End Sub

        Private Sub MyBrowser_Resize(ByVal sender As Object, ByVal e As EventArgs)
            webBrowser1.Width = Width - 6
            webBrowser1.Height = Height - topIndent
        End Sub

        Private Sub buttonEdit1_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs)
            If e.KeyCode = Keys.Enter Then LoadNewDocument(buttonEdit1.Text)
        End Sub

        Private Sub SubscribeWebBrowserEvents()
            AddHandler webBrowser1.DocumentTitleChanged, New EventHandler(AddressOf webBrowser1_DocumentTitleChanged)
            AddHandler webBrowser1.CanGoBackChanged, New EventHandler(AddressOf webBrowser1_CanGoBackChanged)
            AddHandler webBrowser1.CanGoForwardChanged, New EventHandler(AddressOf webBrowser1_CanGoForwardChanged)
        End Sub

        Private Sub webBrowser1_CanGoForwardChanged(ByVal sender As Object, ByVal e As EventArgs)
            btnForward.Enabled = webBrowser1.CanGoForward
        End Sub

        Private Sub webBrowser1_CanGoBackChanged(ByVal sender As Object, ByVal e As EventArgs)
            btnBack.Enabled = webBrowser1.CanGoBack
        End Sub

        Private Sub webBrowser1_DocumentTitleChanged(ByVal sender As Object, ByVal e As EventArgs)
            SetTitle(webBrowser1.DocumentTitle)
            buttonEdit1.Text = Convert.ToString(webBrowser1.Url)
        End Sub

        ' Fields...
        Private _Title As String

        Public ReadOnly Property Title As String
            Get
                Return _Title
            End Get
        End Property

        Public Event TitleChanged As EventHandler

        Protected Overridable Sub OnTitleChangedEvent()
            RaiseEvent TitleChanged(Me, EventArgs.Empty)
        End Sub

        Private Sub SetTitle(ByVal s As String)
            _Title = s
            OnTitleChangedEvent()
        End Sub

        Private Sub LoadNewDocument(ByVal address As String)
            If String.IsNullOrEmpty(address) Then Return
            If address.Equals("about:blank") Then Return
            If Not address.StartsWith("http://") AndAlso Not address.StartsWith("https://") Then
                address = "http://" & address
            End If

            Try
                webBrowser1.Navigate(New Uri(address))
                buttonEdit1.Text = address
            Catch __unusedUriFormatException1__ As UriFormatException
                Return
            End Try
        End Sub

        Private Sub buttonEdit1_ButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.ButtonPressedEventArgs)
            LoadNewDocument(buttonEdit1.Text)
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            webBrowser1.GoBack()
        End Sub

        Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs)
            webBrowser1.GoForward()
        End Sub

        Private Sub simpleButton1_Click_1(ByVal sender As Object, ByVal e As EventArgs)
            LoadNewDocument(buttonEdit1.Text)
        End Sub
    End Class
End Namespace
